docs/DOCKER.md

```diff
@@ -1,11 +1,126 @@
-# How to run the service in docker
+# Running Connectors Service in Docker
 
-Make sure you have the latest `docker` command, then build the image:
+Running Connectors Service in Docker is really easy. All it requires is having Docker installed on a local system, and nothing more!
 
-    make docker-build
+This guide uses generally-available unix commands to demonstrate how to run the Connectors Service in Docker, for Windows you might have to run them in [Unix Subsystem](https://learn.microsoft.com/en-us/windows/wsl/about), rewrite the commands in PowerShell or do the steps in the guide using good old mouse and keyboard :)
 
+## Steps to properly run Connector Service in Docker
 
-To use it, tweak your [configuration file](./CONFIG.md) and run:
+1. Create a directory that will be mounted into the Docker image.
 
-    make docker-run
+This directory will contain the configuration file that will be used to run the Connectors Service. As an example in this guide I will use home directory of the user I'm logged in as.
 
+```sh
+cd ~ && mkdir connectors-python-config
+```
+
+2. Download a sample configuration file from this repository into newly created directory.
+
+Download can be done manually, or just using the line below - make sure to update `--output` argument value if your directory name is different or you want to use a different config file name.
+
+```sh
+curl https://raw.githubusercontent.com/elastic/connectors-python/main/config.yml --output ~/connectors-python-config/config.yml
+```
+
+3. Update the configuration downloaded to make it work for your setup.
+
+In case you're running Connector Service against a dockerised version of Elasticsearch and Kibana, your config file might look the following:
+
+```
+elasticsearch:
+  host: http://host.docker.internal:9200
+  username: elastic
+  password: <YOUR_PASSWORD>
+  ssl: true
+  bulk:
+    queue_max_size: 1024
+    queue_max_mem_size: 25
+    display_every: 100
+    chunk_size: 1000
+    max_concurrency: 5
+    chunk_max_mem_size: 5
+    concurrent_downloads: 10
+  request_timeout: 120
+  max_wait_duration: 120
+  initial_backoff_duration: 1
+  backoff_multiplier: 2
+  log_level: info
+
+service:
+  idling: 30
+  heartbeat: 300
+  max_errors: 20
+  max_errors_span: 600
+  max_concurrent_syncs: 1
+  job_cleanup_interval: 300
+  log_level: INFO
+
+native_service_types:
```

Comment:
Running the connector service in native mode should not be the default for on-prem - I remember we had a GH issue (not finding it atm though) to let users focus on running connector clients via _Build a connector_. 

Could we provide this example with `service_type` so that the default is to run connector clients via _Build a connector_?

Reply:
Correct, let's focus on _Build a connector_ for on-prem. A short note about running the connector in native mode should be enough.

Reply:
I've updated the file, can you take a look @danajuratoni?

I've included a whole config for native connectors, though, should I drop it as well?

Reply:
This section of config file is now for native connectors (for on prem, look at the section above).

Do you mean I leave the whole config for native connectors out, just mentioning that the config downloaded from our repo is native by default?

Reply:
The connector client config is commented out:
https://github.com/elastic/connectors-python/blob/0718287634ccdcc0da8407535b84aa3598570d6f/config.yml#L44-L46

I'm suggesting we also comment out `native_service_types`:
https://github.com/elastic/connectors-python/blob/0718287634ccdcc0da8407535b84aa3598570d6f/config.yml#L29-L41

For both native mode and connector client mode, you don't have to give the example of the whole config file in the MD.

E.g. 

```markdown
3. Update the configuration downloaded to make it work for your [on-prem connector](https://www.elastic.co/guide/en/enterprise-search/current/build-connector.html#build-connector-usage)

Uncomment the `connector_id` and `service_type`:

# connector information
connector_id: <CONNECTOR_ID_FROM_KIBANA>
service_type: '<DESIRED_SERVICE_TYPE>'
```

```markdown
In case you want to run connector service in native mode, you will need slightly different configuration file, having `native_service_types` config option populated instead of `connector_id` and `service_type`:

# remove entries from this list to not run these connectors in this instance of the service
native_service_types:
  - mongodb
  - mysql
```

Reply:
Separate sections look good.
Small remark on the list of native connectors
- I'm not aware that we've tested s3, google_cloud_storage, azure_blob_storage, postgresql, oracle, mssql in native mode and they were working. Considering all the dependencies (drivers etc) are we confident we can support them running in native mode?
- We should not advertise connectors that are released after 8.7 or not at all: dir, sharepoint, jira. Can we take them out of the native connector list?

Reply:
>I'm not aware that we've tested s3, google_cloud_storage, azure_blob_storage, postgresql, oracle, mssql in native mode and they were working. Considering all the dependencies (drivers etc) are we confident we can support them running in native mode?

No, they won't. This `config.yml` is more like the default configuration, which is supposed to be used for development. The real config used in Cloud deployment is [here](https://github.com/elastic/ent-search/blob/bfac9c6d206febc5e94ec169b1ec250d553f867a/dist/shared/cloud/coservices/python-connectors.yml#L37-L38). And you can see only `mysql` is supported for now.

Reply:
Indeed, but this config can be used by on-prem developers. Should we also drop them from the config file, keeping only same ones that are supported in cloud?

Reply:
> this config can be used by on-prem developers

This is correct, in 8.7 customers with Premium license should be able to run native connectors on-prem. That was the premise of my previous comment. So the question would be 
> I'm not aware that we've tested s3, google_cloud_storage, azure_blob_storage, postgresql, oracle, mssql in native mode _on-prem_ ... are we confident we can support them running in native mode?

Reply:
Just checked with @sphilipse and the only native connectors supported (hardcoded) in 8.7 Kibana are MongoDB and MySQL. I'd say it's worth documenting this.

Reply:
For now this part is annoyingly dependant on our functional tests. They are built in the way that the connectors are treated as native, so during improvements of connectors we will make it possible to remove entries from `native_connectors` list, but for now they'll have to stay.

We can remove them from the example config in this doc OR write a separate note saying that the list of actually supported native connectors might differ (we might have to update this list every release or choose a vague description).

What do you think is best?